### PR TITLE
Augmentation of Image Manager 

### PR DESCRIPTION
**Pull request by fklasen**
This is the new PR for the IM example.

### CODE REVIEW

File: edgekv/examples/augment-image-manager-with-ew-edgekv/readme.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Augmenting Image Manager with EdgeWorkers and EdgeKV
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/augment-image-manager-with-ew-edgekv/readme.md:1]: done

File: edgekv/examples/augment-image-manager-with-ew-edgekv/pristine-ew/main.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import {EdgeKV} from './edgekv.js'; //include this file from the parent repository. https://github.com/akamai/edgeworkers-examples/blob/master/edgekv/lib/edgekv.js
+import {logger} from 'log';
+export function onClientResponse(request, response) {
+	if (response.status === 200) {
+    	var key=request.getVariable('PMUSER_PATH_SHA1');
+    	var size=response.getHeader('Content-Length')[0];
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/augment-image-manager-with-ew-edgekv/pristine-ew/main.js]: done

File: edgekv/examples/augment-image-manager-with-ew-edgekv/pristine-ew/main.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import {EdgeKV} from './edgekv.js'; //include this file from the parent repository. https://github.com/akamai/edgeworkers-examples/blob/master/edgekv/lib/edgekv.js
+import {logger} from 'log';
+export function onClientResponse(request, response) {
+	if (response.status === 200) {
+    	var key=request.getVariable('PMUSER_PATH_SHA1');
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/augment-image-manager-with-ew-edgekv/pristine-ew/main.js]: done

File: edgekv/examples/augment-image-manager-with-ew-edgekv/pristine-ew/main.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import {EdgeKV} from './edgekv.js'; //include this file from the parent repository. https://github.com/akamai/edgeworkers-examples/blob/master/edgekv/lib/edgekv.js
+import {logger} from 'log';
+export function onClientResponse(request, response) {
+	if (response.status === 200) {
+    	var key=request.getVariable('PMUSER_PATH_SHA1');
+    	var size=response.getHeader('Content-Length')[0];
+    	var edgeKv = new EdgeKV({namespace: "im", group: "pristine"});
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/augment-image-manager-with-ew-edgekv/pristine-ew/main.js]: done

File: edgekv/examples/augment-image-manager-with-ew-edgekv/derivative-ew/main.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import {EdgeKV} from './edgekv.js'; //include this file from the parent repository. https://github.com/akamai/edgeworkers-examples/blob/master/edgekv/lib/edgekv.js
+import {logger} from 'log';
+export async function onClientRequest(request) {
+	var edgeKv = new EdgeKV({namespace: "im", group: "pristine"});
+	var size = 0;
+	try {
+		var key=request.getVariable('PMUSER_PATH_SHA1');
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/augment-image-manager-with-ew-edgekv/derivative-ew/main.js:7]: done

File: edgekv/examples/augment-image-manager-with-ew-edgekv/derivative-ew/main.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import {EdgeKV} from './edgekv.js'; //include this file from the parent repository. https://github.com/akamai/edgeworkers-examples/blob/master/edgekv/lib/edgekv.js
+import {logger} from 'log';
+export async function onClientRequest(request) {
+	var edgeKv = new EdgeKV({namespace: "im", group: "pristine"});
+	var size = 0;
+	try {
+		var key=request.getVariable('PMUSER_PATH_SHA1');
+	   	var value=await edgeKv.getText({item: key}); 
+	    if(value!==null){
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/augment-image-manager-with-ew-edgekv/derivative-ew/main.js:9]: done

File: edgekv/examples/augment-image-manager-with-ew-edgekv/derivative-ew/main.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import {EdgeKV} from './edgekv.js'; //include this file from the parent repository. https://github.com/akamai/edgeworkers-examples/blob/master/edgekv/lib/edgekv.js
+import {logger} from 'log';
+export async function onClientRequest(request) {
+	var edgeKv = new EdgeKV({namespace: "im", group: "pristine"});
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/augment-image-manager-with-ew-edgekv/derivative-ew/main.js:4]: done

File: edgekv/examples/augment-image-manager-with-ew-edgekv/derivative-ew/main.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import {EdgeKV} from './edgekv.js'; //include this file from the parent repository. https://github.com/akamai/edgeworkers-examples/blob/master/edgekv/lib/edgekv.js
+import {logger} from 'log';
+export async function onClientRequest(request) {
+	var edgeKv = new EdgeKV({namespace: "im", group: "pristine"});
+	var size = 0;
+	try {
+		var key=request.getVariable('PMUSER_PATH_SHA1');
+	   	var value=await edgeKv.getText({item: key});
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/augment-image-manager-with-ew-edgekv/derivative-ew/main.js:8]: done

File: edgekv/examples/augment-image-manager-with-ew-edgekv/pristine-ew/main.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+import {logger} from 'log';
+export function onClientResponse(request, response) {
+	if (response.status === 200) {
+    	var key=request.getVariable('PMUSER_PATH_SHA1');
+    	var size=response.getHeader('Content-Length')[0];
+    	var edgeKv = new EdgeKV({namespace: "im", group: "pristine"});
+     	
+	    edgeKv.putText({item: key, value: size})
+    	.catch(
+			error => logger.log(`EKV: ${error.message}`)
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/augment-image-manager-with-ew-edgekv/pristine-ew/main.js:17]: i think this should be fine, looking at the spec, .message should provide a human readable variant: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/message

[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/augment-image-manager-with-ew-edgekv/pristine-ew/main.js:17]: at the same time, this seems to be very specific to custom messages with specific dependencies, so I'll change it

File: edgekv/examples/augment-image-manager-with-ew-edgekv/derivative-ew/main.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+import {logger} from 'log';
+export async function onClientRequest(request) {
+	var edgeKv = new EdgeKV({namespace: "im", group: "pristine"});
+	var size = 0;
+	try {
+		var key=request.getVariable('PMUSER_PATH_SHA1');
+	   	var value=await edgeKv.getText({item: key}); 
+	    if(value!==null){
+	      size = value;
+	    }
```
[the remark this answers is not in the document]
Comment (fklasen) [reply edgekv/examples/augment-image-manager-with-ew-edgekv/derivative-ew/main.js:11]: fixed, looks good to me

**Comment by fklasen** [comment]
all change request added, tested on production with latest edgekv without any apparent errors

**Review by mryakan — APPROVED** [review]

**Pull request merged**